pangeo_forge/tasks/xarray.py

```diff
@@ -1,43 +1,73 @@
+from typing import List
+
 import fsspec
 import xarray as xr
 from prefect import task
 
 
 @task
-def combine_and_write(sources, target, append_dim, concat_dim, first=True):
+def combine_and_write(
+    sources: List[str], target: str, append_dim: str, concat_dim: str
+) -> List[str]:
     """
-    Combine one or more source datasets into a single `Xarray.Dataset`, then
-    write them to a Zarr store.
+    Write a batch of intermediate files to a combined zarr store.
 
     Parameters
     ----------
-    sources : list of str
-        Path or url to the source files.
+    sources : List[str]
+        A list of URLs pointing to the intermediate files.
     target : str
-        Path or url to the target location of the Zarr store.
+        The URL for the target combined store.
     append_dim : str
         Name of the dimension of which datasets should be appended during write.
     concat_dim : str
-        Name of the dimension of which datasets should be concatenated during read.
-    first : bool
-        Flag to indicate if the target dataset should be expected to already exist.
```

Comment:
This change is also to help out with mapped inputs. It's not clear to me yet if it's a good one, since we're relying on the presence of empty directories to indicate that this is the first write. That seems like a recipe for a race condition.

If possible, it'd be best to have a task explicitly for writing whatever needs to be written on the first write (metadata?) so that all the actual data writes are appends. I'll look into this more later.